DQMOffline/L1Trigger/src/L1TTauOffline.cc

```diff
@@ -0,0 +1,677 @@
+    {
+      edm::LogError("L1TTauOffline") << "invalid collection: reco::MuonCollection " << std::endl;
+      return; 
+    }    
+
+  edm::Handle<reco::BeamSpot> beamSpot;
+  e.getByToken(BsInputTag_, beamSpot);
+
+  if(!beamSpot.isValid())
+    {
+      edm::LogError("L1TTauOffline") << "invalid collection: reco::BeamSpot " << std::endl;
+      return; 
+    }
+
+  edm::Handle<reco::VertexCollection> vertex;
+  e.getByToken(VtxInputTag_, vertex);
+
+  if(!vertex.isValid())
+    {
+      edm::LogError("L1TTauOffline") << "invalid collection: reco::VertexCollection " << std::endl;
+      return; 
+    }
+
+  edm::Handle<l1t::TauBxCollection> l1tCands;				
+  e.getByToken(stage2CaloLayer2TauToken_,l1tCands);
+  
+  if(!l1tCands.isValid())
+    {
+      edm::LogError("L1TTauOffline") << "invalid collection: l1t::TauBxCollection " << std::endl;
+      return;        
+    }
+  
+  edm::Handle<edm::TriggerResults> trigResults;
+  e.getByToken(triggerResults_,trigResults);
+
+  if(!trigResults.isValid())
+    {
+      edm::LogError("L1TTauOffline") << "invalid collection: edm::TriggerResults " << std::endl;
+      return;      
+    }
+  
+  edm::Handle<trigger::TriggerEvent> trigEvent;
+  e.getByToken(triggerEvent_,trigEvent);
+
+  if(!trigEvent.isValid())
+    {
+      edm::LogError("L1TTauOffline") << "invalid collection: trigger::TriggerEvent " << std::endl;
+      return;      
+    }
+
+  edm::Handle<reco::PFMETCollection> mets;
+  e.getByToken(MetInputTag_, mets);
+
+  if(!mets.isValid()) {
+    edm::LogError("L1TTauOffline") << "invalid collection: reco::PFMETCollection " << std::endl;
+    return;
+  }
+
+  eSetup.get<IdealMagneticFieldRecord>().get(m_BField);									
+  const reco::Vertex primaryVertex = getPrimaryVertex(vertex,beamSpot);
+
+  getTightMuons(muons,mets,primaryVertex,trigEvent);
+  getProbeTaus(e,taus,muons,primaryVertex);
+  getTauL1tPairs(l1tCands);
+
+  reco::PFTauCollection::const_iterator tauIt  = taus->begin();
+  reco::PFTauCollection::const_iterator tauEnd = taus->end();
+
+  vector<l1t::Tau> l1tContainer;
+  
+  for (auto tau = l1tCands->begin(0); tau != l1tCands->end(0); ++tau) {
+     l1tContainer.push_back(*tau);
+  }
+
+  vector<l1t::Tau>::const_iterator l1tIt = l1tContainer.begin();;				
+  vector<l1t::Tau>::const_iterator l1tEnd = l1tContainer.end();				
+
+  vector<TauL1TPair>::const_iterator tauL1tPairsIt  = m_TauL1tPairs.begin();
+  vector<TauL1TPair>::const_iterator tauL1tPairsEnd = m_TauL1tPairs.end(); 
+     
+  for(; tauL1tPairsIt!=tauL1tPairsEnd; ++tauL1tPairsIt) {
+
+    float eta = tauL1tPairsIt->eta();
+    float phi = tauL1tPairsIt->phi();
+    float pt  = tauL1tPairsIt->pt();
+
+    // unmatched gmt cands have l1tPt = -1.	
+    float l1tPt  = tauL1tPairsIt->l1tPt();
+
+    vector<int>::const_iterator l1tPtCutsIt  = m_L1tPtCuts.begin();
+    vector<int>::const_iterator l1tPtCutsEnd = m_L1tPtCuts.end();
+
+    int counter = 0;
+
+    for (auto threshold : tauEfficiencyThresholds_) 
```

Comment:
ah- these are not strings, they are doubles - it would be better just to make the map keys ints and make tauEfficiencyThresholds_ a vector of ints.